Core/src/com/serotonin/m2m2/db/H2Proxy.java

```diff
@@ -127,127 +108,82 @@ private Path getDbPathFromUrl(String propertyPrefix) {
-                    }
-                    
                     //Might as well do a compaction here
                     stat.execute("shutdown compact");
                     stat.close();
                 }
                 
-                deleteTempFiles = true;
-                saveProperties = true;
-            }catch(Exception e) {
-                throw new ShouldNeverHappenException(e);
-            }finally {
-                LOG.info("Cleaning up H2 initializataion tests...");
-                if(deleteTempFiles) {
-                    try {
-                        Files.deleteIfExists(dumpPath);
-                    } catch (IOException e) {
-                        LOG.warn("Unable to delete un-necessary H2 dump file " + dumpPath.toString(), e);
-                    }
-                    
-                    try {
-                        Files.deleteIfExists(bakDbPath);
-                    } catch (IOException e) {
-                        LOG.warn("Unable to delete un-necessary H2 backup file " + bakDbPath.toString(), e);
-                    }
-                }
-                if(saveProperties) {
-                    try(FileOutputStream out = new FileOutputStream(h2VersionPath.toFile())) {
-                        h2Properties.store(out, null);
-                    } catch (IOException e) {
-                        LOG.warn("Problem saving H2 version to file " + h2VersionPath.toString(), e);
-                    }
+                try {
+                    LOG.info("Cleaning up H2 initializataion tests...");
+                    Files.deleteIfExists(dumpPath);
+                } catch (IOException e) {
+                    LOG.warn("Unable to delete un-necessary H2 dump file " + dumpPath.toString(), e);
                 }
+                
+            } catch(Exception e) {
+                throw new ShouldNeverHappenException(e);
             }
         }
     }
 
     private String getUrl(String propertyPrefix) {
         String url = Common.envProps.getString(propertyPrefix + "db.url");
         url = StringUtils.replaceMacros(url, System.getProperties());
+        
+        Path dbPath = getDbPathFromUrl(url);
+        
+        //Determine the version info
+        Path dbFolder = dbPath.getParent();
+        String[] matchingDbs = dbFolder.toFile().list((dir, filename) ->{
+            if(filename.startsWith(dbPath.getFileName().toString()))
+                return true;
+            else
+                return false;
+        });
+        
+        int version = 0;
+        for(String match : matchingDbs) {
+            try {
+                int newVersion = getVersion(match);
+                if(newVersion > version)
+                    version = newVersion;
+            } catch (IOException e) {
+                //Ignore
+            }
+        }
+        
+        if(version == 0)
+            version = getCurrentVersion();
```

Comment:
This logic seems funny. I think it may be correct, since presumably the DBs without version numbers would be updated by the upgradePageStore method and we can to create a new DB at the current version's url here in this logic, but it's not so obvious. Maybe it needs a comment if my interpretation is correct?

Reply:
I think you are correct.  I added a comment re-review to see if that is what you were thinking.